sys/net/link_layer/nomac/nomac.c

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (C) 2014 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+#include <errno.h>
+#include <string.h>
+
+#include "netapi.h"
+#include "msg.h"
+#include "netdev/base.h"
+#include "thread.h"
+
+#include "nomac.h"
+
+#ifdef MODULE_NETDEV_DUMMY
+#include "netdev_dummy.h"
+#endif
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#define NOMAC_MSG_QUEUE_SIZE    (16)
+
+static struct {
+    kernel_pid_t registrar_pid; /**< Thread recipient is registered to */
```

Comment:
Ah, I remembered now: `_nomac_recv_cb()` needs the registry to send the data to the upper layers. But since it can't have the registry as a parameter (since it must be of type `netdev_rcv_data_cb_t`) the registry must be global. With this option remaining storing the registrar in the global registry allows for multiple `nomac` threads, since it helps to identify the threads registered to the calling thread of `_nomac_recv_cb()`.